src/create_shards.c

```diff
@@ -417,21 +434,49 @@ ParseWorkerNodeFile(char *workerNodeFilename)
 			}
 		}
 
+		/* if the entire line is whitespace or a comment, skip it */
 		if (*linePointer == '\0' || *linePointer == '#')
 		{
 			continue;
 		}
 
-		/* parse out the node name and node port */
-		parsedValues = sscanf(workerNodeLine, workerLinePattern, nodeName, &nodePort);
-		if (parsedValues != 2)
+		/* parse line; node name is required, but port is optional */
+		fieldCount = sscanf(linePointer, workerLinePattern, nodeName, nodePortString);
+
+		/* adjust field count for zero based indexes */
+		fieldCount--;
+
+		/* raise error if no fields were assigned */
+		if (fieldCount < workerNameIndex)
+		{
+			lineIsInvalid = true;
+		}
+
+		/* no special treatment for nodeName: already parsed by sscanf */
+
+		/* if a second token was specified, convert to integer port */
+		if (fieldCount >= workerPortIndex && *nodePortString != '#')
```

Comment:
Can the following be true with the latest changes?  *nodePortString == '#'
I think it is not. If so, should we remove that check?